### ISSUE TITLE
Keep track of pending tasks in debug mode

### ISSUE THREAD

**Issue by bhalevy** [scylladb/seastar#791]
As seen in https://github.com/scylladb/scylla/issues/7183
In debug build, when using the default allocator it is hard to find pending tasks that are waiting on futures to become ready and not scheduled to run.

One way we can deal with it is keep track of tasks via an intrusive list accessible via a static thread local variable.

**Comment by bhalevy** [comment]
Let's control this functionality with a config option that can be enabled/disabled by the application.

**Comment by denesb** [comment]
I don't think we can make this runtime configurable as that would mean we always pay the overhead of two pointers.

**Comment by bhalevy** [comment]
> I don't think we can make this runtime configurable as that would mean we always pay the overhead of two pointers.

Right. I was thinking about build level configurability.
Then scylla-gdb.py can probe to see if e.g. `thread_local_pending_tasks` is available and if it is, consult it for `scylla tasks`.

**Comment by denesb** [comment]
I think we can unconditionally enable it when `#ifndef SEASTAR_DEFAULT_ALLOCATOR`. We don't use the default allocator for any production build modes.

**Comment by bhalevy** [comment]
cc @avikivity @tgrabiec

**Comment by avikivity** [comment]
We have SEASTAR_TASK_BACKTRACE
